llvm/lib/Transforms/IPO/DynUntrustedAllocPost.cpp

```diff
@@ -341,12 +459,49 @@ class DynUntrustedAllocPost : public ModulePass {
     auto reallocHook = M.getFunction("reallocHook");
     auto deallocHook = M.getFunction("deallocHook");
 
+    // Working under the assumption that all missed cases of Hook calls is 
+    // due to the blocks containing them no longer being reachable, we remove
```

Comment:
Seems like these functions are all the same, exept the instruction list the operate on is a bit different, as well as their debug message and output. maybe should be a function or lambda?

Reply:
Made a function to remove redundancy as suggested.